src/main/lang/Nix.flex

```diff
@@ -2,53 +2,33 @@ package org.nixos.idea.lang;
 
 import com.intellij.lexer.FlexLexer;
 import com.intellij.psi.tree.IElementType;
-import it.unimi.dsi.fastutil.longs.Long2IntMap;
-import it.unimi.dsi.fastutil.longs.Long2IntOpenHashMap;
-import it.unimi.dsi.fastutil.longs.LongArrayList;
-import it.unimi.dsi.fastutil.longs.LongList;
+import it.unimi.dsi.fastutil.ints.AbstractIntList;
+import it.unimi.dsi.fastutil.ints.IntArrayList;
 
 import static org.nixos.idea.psi.NixTypes.*;
 
 %%
 
 %{
-  private final LongList states = new LongArrayList();
-  private final Long2IntMap stateIndexMap = new Long2IntOpenHashMap();
+  private final AbstractIntList states = new IntArrayList();
 
-  {
-    states.add(YYINITIAL);
-    stateIndexMap.put(YYINITIAL, 0);
+  private void pushState(int newState) {
+      // store current state on the stack to allow restoring it in popState(...)
+      states.push(yystate());
+      yybegin(newState);
   }
```

Comment:
nit: We could add a precondition check in this method that verifies that you are not pushing `YYINITIAL`. (Pushing `YYINITIAL` would break the restartability.)

Reply:
Good point. Fixed by 88207134ed4254e7446f1218e687d3982306fe14.